src/components/map/MapDirective.js

```diff
@@ -79,6 +88,49 @@ goog.require('ga_permalink');
 
             map.setTarget(element[0]);
 
+            scope.$watch('::ol3d', function(ol3d) {
+              if (ol3d) {
+                var camera = ol3d.getCesiumScene().camera;
+                var params = gaPermalink.getParams();
+
+                // initial location based on query params
+                var position, heading, pitch;
+                if (params.lon && params.lat && params.elevation) {
+                  var lon = parseFloat(params.lon);
+                  var lat = parseFloat(params.lat);
+                  var elevation = parseFloat(params.elevation);
+                  position = new Cesium.Cartographic(lon, lat, elevation);
+                }
+                if (params.heading) {
+                  heading = Cesium.Math.toRadians(parseFloat(params.heading));
+                }
+                if (params.pitch) {
+                  pitch = Cesium.Math.toRadians(parseFloat(params.pitch));
+                }
+                camera.setView({
+                  positionCartographic: position,
+                  heading: heading,
+                  pitch: pitch,
+                  roll: 0.0
+                });
+
+                // update permalink
+                camera.moveEnd.addEventListener(function() {
+                  // remove 2d params
+                  gaPermalink.deleteParams(['X', 'Y', 'zoom']);
```

Comment:
Super minor, but I'd have used three calls to `deleteParam` to avoid creating an array and adding a `deleteParams` function to `gaPermalink`.